### DISCUSSION TITLE
I can't get content through the console.

### DISCUSSION THREAD

**Discussion by isgarat** [community/community#133942] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi,  I need help with the following...
When I want to download script from ssh console, either with wget or curl, at no time does it download, if from the same network, through a web browser, I can download the raw content.
When I execute the curl command it returns the following error: 

```
root@pve:~# curl https://github.com
curl: (35) Recv failure: Connection reset by peer
```
when I execute it with wget it does not generate any error message, it just seems to "do nothing".
I am executing the wget or curl commands from a vm in proxmox, if from the vm I execute a ping to github.com the IP returns correctly. But wget and curl commands don't work.

what could be the problem? or how can I solve?
kind regards

**Comment by Sivamani-18** [comment]
The issue you're experiencing with `curl` and `wget` not being able to download scripts from the SSH console, while the same operation works from a web browser, suggests a network or configuration problem. Here are several potential causes and solutions:

1. **Network Configuration or Firewall:**
   - **Outbound Traffic Blocking:** Ensure that there are no firewall rules or network policies blocking outbound HTTPS connections from the VM. Sometimes, network administrators configure firewalls to block certain types of traffic for security reasons.
   - **Proxy Configuration:** If your network uses a proxy server, you may need to configure `curl` and `wget` to use it. You can set the proxy in the environment variables:
     ```sh
     export http_proxy=http://proxy_server:port
     export https_proxy=http://proxy_server:port
     ```
   - **NAT Configuration:** Check if the Proxmox host has correct NAT settings and that the VM has proper network access.

2. **SSL/TLS Configuration:**
   - **Certificates:** Ensure that your VM has up-to-date CA certificates. You can update CA certificates on Debian-based systems with:
     ```sh
     apt-get update
     apt-get install ca-certificates
     ```
   - **SSL/TLS Versions:** Some servers require newer TLS versions. Ensure `curl` and `wget` are configured to use modern TLS:
     ```sh
     curl --tlsv1.2 https://github.com
     ```

3. **DNS Resolution:**
   - Even though `ping` works, it's worth checking if `curl` and `wget` use the same DNS resolution. Ensure that `/etc/resolv.conf` has the correct DNS servers listed.

4. **Check Logs and Errors:**
   - For `curl`, you can use the `-v` (verbose) flag to get more detailed output:
     ```sh
     curl -v https://github.com
     ```
   - For `wget`, use the `-d` (debug) flag:
     ```sh
     wget -d https://github.com
     ```

5. **Updates and Compatibility:**
   - Ensure that `curl` and `wget` are up to date. Older versions may have compatibility issues with modern SSL/TLS implementations:
     ```sh
     apt-get update
     apt-get upgrade curl wget
     ```

6. **Local Configuration Issues:**
   - Sometimes, user-specific configurations in `.curlrc` or `.wgetrc` files can cause issues. Check if these files exist in your home directory and try renaming them temporarily:
     ```sh
     mv ~/.curlrc ~/.curlrc.bak
     mv ~/.wgetrc ~/.wgetrc.bak
     ```

7. **Testing Alternative URLs:**
   - Try downloading from another secure URL to see if the issue is specific to GitHub:
     ```sh
     curl https://www.google.com
     wget https://www.google.com
     ```

8. **Proxy Bypass for Local Network:**
   - If you have a proxy set but want to bypass it for local addresses, ensure you set the `no_proxy` variable correctly:
     ```sh
     export no_proxy=localhost,127.0.0.1,.localdomain
     ```

**Reply by isgarat** [reply]
Hello, I am going to verify everything mentioned in this thread, after that, I am going to present the results.
